executor/join_result_generators.go

```diff
@@ -157,21 +199,25 @@ type leftOuterSemiJoinResultGenerator struct {
 
 // emitMatchedInners implements joinResultGenerator interface.
 func (outputer *leftOuterSemiJoinResultGenerator) emitMatchedInners(outer Row, inners []Row, resultBuffer []Row) ([]Row, bool, error) {
+	if len(inners) == 0 {
```

Comment:
extract len(inners) and len(outers) as variables will be more readable
so as the others.